File: BackendAPI/Application/DTOs/Output/VeilingKlokOutputDto.cs
using Domain.Enums;

namespace Application.DTOs.Output;

public class VeilingKlokOutputDto
{
    public required Guid Id { get; set; }
    public required VeilingKlokStatus Status { get; set; }
    public required int PeakedLiveViews { get; set; }
    public required DateTimeOffset CreatedAt { get; set; }
    public required string RegionOrState { get; set; }
    public required string Country { get; set; }
    public required int CurrentBids { get; set; }
    public required int TotalProducts { get; set; }
    public DateTimeOffset? ScheduledAt { get; set; }
    public DateTimeOffset? StartedAt { get; set; }
    public DateTimeOffset? EndedAt { get; set; }

    public decimal? HighestBidAmount { get; set; }
    public decimal? LowestBidAmount { get; set; }
    public required List<ProductOutputDto> Products { get; set; }
}

File: FrontendWeb/src/declarations/dtos/output/VeilingKlokOutputDto.ts
import {VeilingKlokStatus} from '../../enums/VeilingKlokStatus';
import {ProductOutputDto} from './ProductOutputDto';

// VeilingKlokOutputDto.ts
export interface VeilingKlokOutputDto {
	id: string;
	status: VeilingKlokStatus;
	peakedLiveViews: number;
	createdAt: string;
	regionOrState: string;
	country: string;
	currentBids: number;
	totalProducts: number;
	scheduledAt: string;
	startedAt: string | null;
	endedAt: string | null;
	veilingDurationSeconds: number;
	highestBidAmount: number | null;
	lowestBidAmount: number | null;
	products: ProductOutputDto[];
}
